### PR TITLE
Feature/global exception handler

### PR DESCRIPTION
**Pull request by Anatolii-Omelchenko**

### CODE REVIEW

File: src/main/java/baza/trainee/exceptions/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RequiredArgsConstructor
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/exceptions/GlobalExceptionHandler.java:16]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/exceptions/custom/BasicApplicationException.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.http.HttpStatus;
+
+/**
+ * This class serves as a base for creating custom exceptions
+ * in the application. It extends the RuntimeException class and
+ * holds information about the error's HTTP status.
+ */
+@Getter
+public class BasicApplicationException extends RuntimeException {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/exceptions/custom/BasicApplicationException.java:14]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/exceptions/custom/EntityNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package baza.trainee.exceptions.custom;
+
+import org.springframework.http.HttpStatus;
+
+public class EntityNotFoundException extends BasicApplicationException {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/exceptions/custom/EntityNotFoundException.java:7]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/exceptions/custom/MethodArgumentNotValidException.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package baza.trainee.exceptions.custom;
+
+import org.springframework.http.HttpStatus;
+
+public class MethodArgumentNotValidException extends BasicApplicationException {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/exceptions/custom/MethodArgumentNotValidException.java:7]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/exceptions/custom/NullEntityReferenceException.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package baza.trainee.exceptions.custom;
+
+import org.springframework.http.HttpStatus;
+
+public class NullEntityReferenceException extends BasicApplicationException {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/exceptions/custom/NullEntityReferenceException.java:7]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/utils/ControllerUtils.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package baza.trainee.utils;
+
+import baza.trainee.exceptions.custom.MethodArgumentNotValidException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+
+import java.util.List;
+
+public class ControllerUtils {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/utils/ControllerUtils.java:11]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/utils/LoggingService.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package baza.trainee.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class LoggingService {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/utils/LoggingService.java:10]: It must be an empty string before Javadoc comments.

File: src/main/java/baza/trainee/exceptions/GlobalExceptionHandler.java (hunk 1 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/exceptions/GlobalExceptionHandler.java:16]: fixed

File: src/main/java/baza/trainee/exceptions/custom/BasicApplicationException.java (hunk 2 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/exceptions/custom/BasicApplicationException.java:14]: fixed

File: src/main/java/baza/trainee/exceptions/custom/EntityNotFoundException.java (hunk 3 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/exceptions/custom/EntityNotFoundException.java:7]: fixed

File: src/main/java/baza/trainee/exceptions/custom/NullEntityReferenceException.java (hunk 5 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/exceptions/custom/NullEntityReferenceException.java:7]: fixed

File: src/main/java/baza/trainee/utils/ControllerUtils.java (hunk 6 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/utils/ControllerUtils.java:11]: fixed

File: src/main/java/baza/trainee/utils/LoggingService.java (hunk 7 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/utils/LoggingService.java:10]: fixed

**Review by Olga232 — APPROVED** [review]

**Review by malyshevhen — APPROVED** [review]

**Pull request merged**